applications/hotornot/src/hon_util.erl

```diff
@@ -20,59 +21,82 @@
--spec candidate_rates(ne_binary(), binary()) ->
+-spec candidate_rates(ne_binary(), api_binary(), ne_binary(), binary()) ->
                              {'ok', wh_json:objects()} |
                              {'error', atom()}.
-candidate_rates(ToDID) ->
-    candidate_rates(ToDID, <<>>).
-candidate_rates(ToDID, FromDID) ->
+candidate_rates(AccountId, Direction, ToDID) ->
+    candidate_rates(AccountId, Direction, ToDID, <<>>).
+candidate_rates(AccountId, Direction, ToDID, FromDID) ->
     E164 = wnm_util:to_e164(ToDID),
-    find_candidate_rates(E164, FromDID).
+    find_candidate_rates(AccountId, Direction, E164, FromDID).
+
+find_candidate_rates(AccountId, Direction, E164, _FromDID) when byte_size(E164) > ?MIN_PREFIX_LEN ->
+    Keys = build_keys(Direction, E164),
+    RateDbs = rate_dbs(AccountId),
+    case lists:foldl(fun acc_candidate_rates/2, {'ok', Keys, 0, []}, RateDbs) of
+        {'ok', _, _, Rates} -> {'ok', Rates};
+        E -> E
+    end;
+find_candidate_rates(_, _, DID, _) ->
+    lager:debug("DID ~s is too short", [DID]),
+    {'error', 'did_too_short'}.
 
-find_candidate_rates(E164, _FromDID) when byte_size(E164) > ?MIN_PREFIX_LEN ->
-    Keys = build_keys(E164),
+build_keys(Direction, <<"+", E164/binary>>) ->
+    build_keys(Direction, E164);
+build_keys(Direction, <<D:1/binary, Rest/binary>>) ->
+    build_keys(Direction, Rest, D, [[D, Direction]]).
+
+build_keys(Direction, <<D:1/binary, Rest/binary>>, Prefix, Acc) ->
+    build_keys(Direction, Rest, <<Prefix/binary, D/binary>>, [[<<Prefix/binary, D/binary>>, Direction] | Acc]);
+build_keys(_, <<>>, _, Acc) -> Acc.
+
+-spec rate_dbs(ne_binary()) -> list().
+rate_dbs(AccountId) ->
+    {ok, A} = couch_mgr:open_doc(?WH_ACCOUNTS_DB, AccountId),
+    lists:foldl(fun acc_rate_dbs/2, [], lists:reverse([AccountId | kz_account:tree(A)]))
+    ++ [?WH_RATES_DB].
+
+-spec acc_rate_dbs(ne_binary(), list()) -> list().
+acc_rate_dbs(AccountId, Acc) ->
+    Db = wh_util:format_account_db(<<AccountId/binary, "-", ?WH_RATES_DB/binary>>),
+    case couch_mgr:db_exists(Db) of
+        true -> [Db | Acc];
+        false -> Acc
+    end.
 
-    lager:debug("searching for prefixes for ~s: ~p", [E164, Keys]),
-    case couch_mgr:get_results(?WH_RATES_DB, <<"rates/lookup">>, [{'keys', Keys}
-                                                                  ,'include_docs'
-                                                                 ])
-    of
-        {'ok', []}=OK -> OK;
+-spec acc_candidate_rates(ne_binary(), {'ok', list(), integer(), list()} | {'error', any()}) ->
+    {'ok', list(), integer(), list()} | {'error', any()}.
+acc_candidate_rates(RateDb, {'ok', Keys, Lvl, Acc}) ->
+    case couch_mgr:get_results(RateDb, <<"rates/lookup">>, [{'keys', Keys}, 'include_docs']) of
         {'ok', ViewRows} ->
-            {'ok', [wh_json:get_value(<<"doc">>, ViewRow) || ViewRow <- ViewRows]};
+            {'ok', Keys, Lvl + 1, Acc ++ [normalize_rate(Rate, Lvl) || Rate <- ViewRows]};
```

Comment:
@jamesaimonetti 
This code is just simplier for testing.
It's not too hard to rewrite it.
I'll wait for k-anderson opinion about way and strategies. :)